### DISCUSSION TITLE
Upload button

### DISCUSSION THREAD

**Discussion by AcernisSupport** [ant-design-blazor/ant-design-blazor#4360] [General]
Good evening list 
(at least in Italy it's night)

I'm trying to use the UPLOAD control to import a file and save it in a folder so that I can then process the xlsx file to import the data

I tried but I can't figure out how I can save the file

I am convinced that it can be done.

Can someone help me with an example so I can study and implement it?

Thank you and good work

**Comment by ElderJames** [comment]
Thanks for contacting us @AcernisSupport , you can create a endpoint for fetch the uploading files.

```
app.MapPost("/upload", async (IFormFile file) =>
{
    var filePath = Path.Combine(app.Environment.WebRootPath, "uploads");
    if (!Directory.Exists(filePath))
    {
        Directory.CreateDirectory(filePath);
    }
    var uploads = Path.Combine(filePath, file.FileName);
    using var fileStream = File.OpenWrite(uploads);
    using var uploadStream = file.OpenReadStream();
    await uploadStream.CopyToAsync(fileStream);

    var uploadsPath = $"/uploads/" + file.FileName;

    return Results.Ok(new { url = uploadsPath });
}).DisableAntiforgery();
```

Then setup the Upload component whit the action '/upload'

```
<Upload Action="/upload" >

</Upload>

```

**Comment by AcernisSupport** [comment]
Hello ElderJames

First of all, thank you for your answer and sorry for the delay in my answer, but between influence and work I was only able to see the code today.

The first part I figured out where to insert it but the second part I wouldn't be like using it in code.

I point out that I'm trying to use the UPDATE control that exists in the documentation 
(
`<Upload Action="https://www.mocky.io/v2/5cc8019d300000980a055e76"
        Name="files"
        OnSingleCompleted="OnSingleCompleted">
    <Button Icon="upload">
        <span>Click to Upload</span>
    </Button>
</Upload>`
 ) 
but it always gives me a 405 error (even if I enter http://localhost:5000/updates or http://localhost:5000/wwwroots/updates. I created the folder under wwwroots being in debug)

I don't know how to manage it anymore and it's frustrating (certainly my fault) not to succeed in something that I think is so commonly used.

Looking forward to your help/suggestion, I send you cordial greetings and wishing you a happy new year

Luca Perini

**Comment by ElderJames** [comment]
Hello @AcernisSupport ，did you add an API for uploading files? Then you need to set the `Action` parameter as the API url.

**Comment by AcernisSupport** [comment]
Hi ElderJames

No, I didn't add any API


Now that I understand that an API was needed, now I added it and now it works

Thank you and good work
